src/System.Private.CoreLib/src/System/Reflection/RuntimeConstructorInfo.cs

```diff
@@ -372,7 +372,7 @@ public override MethodBody GetMethodBody()
         {
             MethodBody mb = RuntimeMethodHandle.GetMethodBody(this, ReflectedTypeInternal);
             if (mb != null)
-                mb.m_methodBase = this;
+                mb._methodBase = this;
```

Comment:
Change the return type of `RuntimeMethodHandle.GetMethodBody` to be `RuntimeMethodBody` to make this work.